aquadoggo/src/materializer/tasks/schema.rs

```diff
@@ -1,9 +1,285 @@
+}
+
+/// Determine the schema of a view.
+async fn get_schema_for_view(
+    view_id: &DocumentViewId,
+    context: &Context,
+) -> Result<SchemaId, TaskError> {
+    let sample_operation_id = view_id.graph_tips().first().unwrap();
+    let sample_operation = context.store.get_operation_by_id(sample_operation_id).await;
+    let sample_operation = sample_operation.map_err(|_| TaskError::Critical)?.unwrap();
+    Ok(sample_operation.operation().schema())
+}
+
+/// Retrieve schema definitions that use the targetted schema field definition as one of their fields.
+async fn get_related_schema_definitions(
+    target_field_definition: &DocumentViewId,
+    context: &Context,
+) -> Result<Vec<DocumentViewId>, TaskError> {
+    // Retrieve all schema definition documents from the store.
+    let schema_definitions = context
+        .store
+        .get_documents_by_schema(&SchemaId::SchemaDefinition(1))
+        .await
+        .map_err(|_err| TaskError::Critical)
+        .unwrap();
+
+    // Collect all schema definitions that use the targetted field definition
+    let mut related_schema_definitions = vec![];
+    for schema in schema_definitions {
+        let fields_value = schema.fields().get("fields").unwrap().value();
+        if let OperationValue::PinnedRelationList(fields) = fields_value {
+            if fields
+                .iter()
+                .any(|field_view_id| &field_view_id == target_field_definition)
+            {
+                related_schema_definitions.push(schema.id().clone())
+            } else {
+                continue;
+            }
+        } else {
+            // Abort if there are schema definitions in the store that don't match the schema
+            // definition schema.
+            Err(TaskError::Critical)?
+        }
+    }
+
+    Ok(related_schema_definitions)
+}
+
+#[cfg(test)]
+mod tests {
+    use log::debug;
+    use std::convert::{TryFrom, TryInto};
+
+    use p2panda_rs::document::DocumentViewId;
+    use p2panda_rs::entry::{sign_and_encode, Entry};
+    use p2panda_rs::identity::{Author, KeyPair};
+    use p2panda_rs::operation::{
+        Operation, OperationEncoded, OperationFields, OperationValue, PinnedRelationList,
+        VerifiedOperation,
+    };
+    use p2panda_rs::schema::{FieldType, Schema, SchemaId, SchemaVersion};
+    use p2panda_rs::storage_provider::traits::{OperationStore, StorageProvider};
+    use p2panda_rs::test_utils::fixtures::create_operation;
+    use rstest::rstest;
+
+    use crate::context::Context;
+    use crate::db::stores::test_utils::{test_db, TestSqlStore};
+    use crate::graphql::client::{EntryArgsRequest, PublishEntryRequest};
+    use crate::materializer::tasks::reduce_task;
+    use crate::materializer::TaskInput;
+    use crate::schema::SchemaProvider;
+    use crate::Configuration;
+
+    use super::schema_task;
+
+    /// Init env logger for tests.
+    fn init() {
+        let _ = env_logger::builder().is_test(true).try_init();
+    }
```

Comment:
Could this be initialised on top-level inside the `tests` module of this project for all tests?

Reply:
This is how env_logger recommends to do it in their [docs](https://docs.rs/env_logger/latest/env_logger/#capturing-logs-in-tests). We could do what you suggest with [ctor](https://docs.rs/ctor/latest/ctor/) though! I would leave it as it is for now and just put that init function in a test utils file if we use it again later.

Reply:
Sounds good!